how2meet/utils.py

```diff
@@ -8,60 +8,60 @@
-    # NOTE: This is a blocking call, so we use an async client
-    async with AsyncClient() as client:
-        events = await client.get(f"{BASE_URL}/api/events/", timeout=10)
-    try:
-        events = events.json()
-    except:
-        events = []
-
-    return events
-
-
-async def get_event_api(event_id: str) -> dict:
-    """Get single event from API"""
-    # NOTE: This is a blocking call, so we use an async client
-    async with AsyncClient() as client:
-        event = await client.get(f"{BASE_URL}/api/events/{event_id}", timeout=10)
-    try:
-        event = event.json()
-    except:
-        event = {}
-
-    return event
-
-
-async def delete_event_api(event_id: str) -> int:
-    """Delete single event from API"""
-    async with AsyncClient() as client:
-        response = await client.delete(f"{BASE_URL}/api/events/{event_id}", timeout=10)
-        response.raise_for_status()
-
-    await ui.run_javascript("location.reload();")  # Refresh page to show changes
-
-    return response.status_code
-
-
-async def post_event_api(event_json) -> int:
-    """
-    Post single event to API
-    TODO: rename to create?
-    """
-    async with AsyncClient() as client:
-        response = await client.post(f"{BASE_URL}/api/events/", data=event_json, timeout=10)
-        response.raise_for_status()
 
-    return response.status_code
+class APIClient:
+    def __init__(self, base_url: str = BASE_URL) -> None:
+        self.base_url = base_url
+
+    @classmethod
+    async def get_events(cls) -> list:
+        """
+        Get list of all events from API
+        TODO: Limit the number of events returned
+        TODO: query params?
+        """
+        # NOTE: This is a blocking call, so we use an async client
+        async with AsyncClient() as client:
+            events = await client.get(f"{BASE_URL}/api/events/", timeout=10)
+        try:
+            events = events.json()
+        except:
+            events = []
+
+        return events
+
+    @classmethod
+    async def get_event(cls, event_id: str) -> dict:
+        """Get single event from API"""
+        # NOTE: This is a blocking call, so we use an async client
+        async with AsyncClient() as client:
+            event = await client.get(f"{BASE_URL}/api/events/{event_id}", timeout=10)
+        try:
+            event = event.json()
+        except:
+            event = {}
+
+        return event
+
+    @classmethod
+    async def delete_event(cls, event_id: str) -> int:
+        """Delete single event from API"""
+        async with AsyncClient() as client:
+            response = await client.delete(f"{BASE_URL}/api/events/{event_id}", timeout=10)
+            response.raise_for_status()
+
+        await ui.run_javascript("location.reload();")  # Refresh page to show changes
+
+        return response.status_code
+
+    @classmethod
+    async def post_event(cls, event_json) -> int:
+        """
+        Post single event to API
+        TODO: rename to create?
```

Comment:
I'd lean towards create over post. It makes it more clear exactly what it's doing.